### PR TITLE
Issue #Sb-6925 fix: env config migration

### PR DESCRIPTION
**Pull request by loganathan1989**

### CODE REVIEW

**Comment by harishgilimi** [comment]
@loganathan1989 
As per process we need test scenarios updated in JIRA to start reviewing this PR and test scenarios should be reviewed with Tech Manager

cc: @bvinayakumar @gsbajaj72

File: src/app/helpers/config/configHelper.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+let cacheRefreshEnabled = false
+const configReqKey = 'instance.portal'
+
+// left side 'key' is the configuration service key and right side 'value' is the envhelper key
+const configMap = {
+  sunbird_autocreate_trampoline_user: 'PORTAL_AUTOCREATE_TRAMPOLINE_USER',
+  sunbird_enabless_permission_check: 'ENABLE_PERMISSION_CHECK',
+  sunbird_instance: 'PORTAL_TITLE_NAME',
+  sunbird_theme: 'PORTAL_THEME',
+  sunbird_portal_default_language: 'PORTAL_DEFAULT_LANGUAGE',
```
Comment (rajeevsathish) [inline src/app/helpers/config/configHelper.js]: Is portal needed as it is not there in any other variable

File: src/app/helpers/config/configHelper.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    let configs = {}
+    if (configServiceEnabled) {
+      readConfigs(configReqKey).then(function (configObj) {
+        console.log('Info: Configurations fetched from API successfully')
+        resolve(checkAndStoreConfigs(configObj))
+      }, function (err) {
+        console.log('Info: Configurations falling back to environment variables due to error in fetching config from API')
+        resolve(checkAndStoreConfigs(configs))
+      })
+      if (cacheRefreshEnabled == false) {
```
Comment (rajeevsathish) [inline src/app/helpers/config/configHelper.js]: Can't we use notof instead of ==false?

File: src/app/helpers/config/configuration.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+__get = function(configKey){
+    return configData[configKey]
+}
+
+__set = function(configKey,configValue){
+    configData[configKey] = configValue
+}
+
+module.exports = {
+    get: __get,
```
Comment (rajeevsathish) [inline src/app/helpers/config/configuration.js]: where is the get and set used?

File: src/app/helpers/environmentVariablesHelper.js [hunk 4]
```diff
@@ -7,7 +7,8 @@ let envVariables = {
   LEARNER_URL: env.sunbird_learner_player_url || 'https://staging.open-sunbird.org/api/',
   CONTENT_URL: env.sunbird_content_player_url || 'https://staging.open-sunbird.org/api/',
   CONFIG_URL: env.sunbird_config_service_url || 'https://staging.open-sunbird.org/api/config/',
-  CONFIG_CACHE_TTL: env.config_cache_ttl || 600,
+  CONFIG_REFRESH_INTERVAL: env.config_refresh_interval || 10,
+  CONFIG_SERVICE_ENABLED: env.config_service_enabled || true,
```
Comment (rajeevsathish) [inline src/app/helpers/environmentVariablesHelper.js]: by Default the value should be False not true

File: src/app/routes/clientRoutes.js [hunk 5]
```diff
@@ -74,33 +74,30 @@ module.exports = (app, keycloak) => {
...
 
 function getLocals (req, callback) {
-  let config_key_allow_signup = 'instance.allow_signup'
-  let env_key_allow_signup = 'ENABLE_SIGNUP'
-
-  configHelper.getConfig(config_key_allow_signup, env_key_allow_signup, function (err, allow_signup) {
-    if (!err) {
+  let config_key_allow_signup = 'ENABLE_SIGNUP'
+  let allow_signup = configHelper.getConfig(config_key_allow_signup)
+    if (allow_signup !== undefined) {
```
Comment (rajeevsathish) [inline src/app/routes/clientRoutes.js]: why is this a undefined check not a true false chk?

File: src/app/server.js [hunk 6]
```diff
@@ -206,20 +208,22 @@ function runApp () {
...
-        .on('response', function (res) {
-          if (res.statusCode === 200) {
-            req.pipe(fs.createWriteStream(path.join(__dirname, 'dist', 'index.ejs')))
-          } else {
-            console.log('Error while fetching '+envHelper.PORTAL_CDN_URL + 'index.' + packageObj.version + '.' + packageObj.buildHash + '.ejs file when CDN enabled');
-          }
-        })
-    }
-    console.log('app running on port ' + envHelper.PORTAL_PORT)
+  // start server after fetching the configuration data
```
Comment (rajeevsathish) [inline src/app/server.js]: what happens if a error is there in the call will the server be started?

File: src/app/helpers/resourceBundles/build.js [hunk 7]
```diff
@@ -4,6 +4,7 @@ const fs = require('fs')
 const properties = require('properties')
 const _ = require('lodash')
 const envHelper = require('./../environmentVariablesHelper.js')
```
Comment (kochhar) [inline src/app/helpers/resourceBundles/build.js]: remove envHelper if not needed

File: src/app/helpers/resourceBundles/index.js [hunk 8]
```diff
@@ -1,5 +1,6 @@
 let dateFormat = require('dateformat')
 const envHelper = require('./../environmentVariablesHelper.js')
```
Comment (kochhar) [inline src/app/helpers/resourceBundles/index.js]: remove emvHelper

File: src/app/helpers/config/configHelper.js [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+const cron = require('node-cron');
+const configServiceEnabled = envHelper.CONFIG_SERVICE_ENABLED
+let configRefreshInterval = envHelper.CONFIG_REFRESH_INTERVAL
+let cacheRefreshEnabled = false
+const configReqKey = 'instance.portal'
+
+// left side 'key' is the configuration service key and right side 'value' is the envhelper key
+const configMap = {
+  sunbird_autocreate_trampoline_user: 'PORTAL_AUTOCREATE_TRAMPOLINE_USER',
+  sunbird_enabless_permission_check: 'ENABLE_PERMISSION_CHECK',
```
[the remark this answers is not in the document]
Comment (loganathan1989) [reply src/app/helpers/config/configHelper.js]: These are env variable names used in envvhelper and i followed the same.Can we use different names?

File: src/app/helpers/config/configHelper.js [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+  sunbird_instance: 'PORTAL_TITLE_NAME',
```
[the remark this answers is not in the document]
Comment (loganathan1989) [reply src/app/helpers/config/configHelper.js]: These are env variable names used in envvhelper and i followed the same.Can we use different names?

File: src/app/helpers/config/configHelper.js (hunk 1 above)
Comment (loganathan1989) [reply src/app/helpers/config/configHelper.js]: These are env variable names used in envvhelper and i followed the same.Can we use different names?

File: src/app/helpers/config/configuration.js (hunk 3 above)
Comment (loganathan1989) [reply src/app/helpers/config/configuration.js]: The exported get and set methods were used in confighelper.js file to set and get configs

File: src/app/helpers/environmentVariablesHelper.js (hunk 4 above)
Comment (loganathan1989) [reply src/app/helpers/environmentVariablesHelper.js]: As per Bajaj advice i set it to true,can we change it to false ?

File: src/app/server.js (hunk 6 above)
Comment (loganathan1989) [reply src/app/server.js]: If there is an error it will fallback to environment variables configuration and this is handled in configHelper.

**Pull request merged**